pysollib/games/hitormiss.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+# -*- mode: python; coding: utf-8; -*-
+# ---------------------------------------------------------------------------
+#
+# Copyright (C) 1998-2003 Markus Franz Xaver Johannes Oberhumer
+# Copyright (C) 2003 Mt. Hood Playing Card Co.
+# Copyright (C) 2005-2009 Skomoroh
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# ---------------------------------------------------------------------------
+
+from pysollib.game import Game
+from pysollib.gamedb import GI, GameInfo, registerGame
+from pysollib.layout import Layout
+from pysollib.mygettext import _
+from pysollib.pysoltk import MfxCanvasText
+from pysollib.stack import \
+        AbstractFoundationStack, \
+        WasteStack, \
+        WasteTalonStack
+from pysollib.util import ANY_SUIT, RANKS, \
+        VARIABLE_REDEALS
+
+
+# ************************************************************************
+# *
+# ************************************************************************
+
+class HitOrMiss_Foundation(AbstractFoundationStack):
+    def acceptsCards(self, from_stack, cards):
+        game = self.game
+        return game.rank == cards[0].rank  # check the rank
+
+    def getHelp(self):
+        return _('Foundation. Place cards here that match the declared rank.')
+
+
+class HitOrMiss_Talon(WasteTalonStack):
+    def canDealCards(self):
+        game = self.game
+        if (game.deadDeals == 2 and len(self.cards) == 0 and
+                self.max_rounds == VARIABLE_REDEALS):
+            return False
+        return True
+
+    def dealCards(self, sound=False):
+        game = self.game
+        old_state = game.enterState(game.S_FILL)
+        game.saveStateMove(2 | 16)  # for undo
+        if len(self.cards) > 0:
+            game.rank = game.rank + 1
```

Comment:
@joeraz : please use `+= 1` here and elsewhere: https://careerkarma.com/blog/python-operator/

Reply:
Done.